cdf_fabric_replicator/extractor.py

```diff
@@ -304,19 +315,52 @@ def get_events(self, df: DataFrame) -> list[EventWrite]:
             events.append(new_event)
         return events
 
-    def convert_lakehouse_data_to_df(self, file_path: str, token: str) -> DataFrame:
+    def retrieve_external_ids_from_lakehouse(
+        self, file_path: str, token: str
+    ) -> list[str]:
         try:
-            dt = DeltaTable(
-                file_path,
-                storage_options={"bearer_token": token, "user_fabric_endpoint": "true"},
+            dt = DeltaTable(file_path, storage_options={"bearer_token": token})
+            table = dt.to_pyarrow_table(columns=["externalId"])
```

Comment:
Will this do the actual fetch of filtered data when the filter is applied?

Reply:
The data will be fetched on line 324 after the unique operation.  The data will also be restricted to the externalId column